packages/uicore/src/components/StepWizard/StepWizard.tsx

```diff
@@ -163,8 +163,9 @@ export function StepWizard<SharedObject = Record<string, unknown>>(
         stepData?.stepIndex === React.Children.toArray(nestedWizard.props.children as any).length
       ) {
         nestedWizard.props.onCompleteWizard(prevStepData)
-        if (state.activeStep === state.totalSteps) {
-          return true
+        if (stepNumber > state.totalSteps) {
+          // Not a valid step stepNumber
+          return false
         }
```

Comment:
Do we want to remove this condition?

Reply:
We can remove this and make one generic condition for all the cases if stepNumber is not valid that is the last else condition. But I am not sure how that will affect other scenarios